internal/sirius/users.go

```diff
@@ -0,0 +1,101 @@
+package sirius
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+type UserStatus string
+
+func (us UserStatus) String() string {
+	return string(us)
+}
+
+func (us UserStatus) TagColour() string {
+	if us == "Suspended" {
+		return "govuk-tag--grey"
+	} else if us == "Locked" {
+		return "govuk-tag--orange"
+	} else {
+		return ""
+	}
+}
+
+type apiUser struct {
+	ID          int    `json:"id"`
+	DisplayName string `json:"displayName"`
+	Surname     string `json:"surname"`
+	Email       string `json:"email"`
+	Locked      bool   `json:"locked"`
+	Suspended   bool   `json:"suspended"`
+}
+
+type User struct {
+	ID          int    `json:"id"`
+	DisplayName string `json:"displayName"`
+	Email       string `json:"email"`
+	Status      UserStatus
+}
+
+func (c *Client) ListUsers(ctx context.Context, cookies []*http.Cookie) ([]User, error) {
+	var apiUsers []apiUser
+
+	req, err := c.newRequest(ctx, http.MethodGet, "/api/v1/users", nil, cookies)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusUnauthorized {
+		return nil, ErrUnauthorized
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.New("returned non-2XX response: " + strconv.Itoa(resp.StatusCode))
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&apiUsers)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var users []User
+
+	sort.SliceStable(apiUsers, func(i, j int) bool {
```

Comment:
Do we have to sort the API response?

Reply:
Yeah, the API returns them in row order, which is completely useless for scanning

Reply:
🤦